windows-agent/internal/proservices/proservices.go

```diff
@@ -85,7 +85,7 @@ func New(ctx context.Context, publicDir, privateDir string, args ...Option) (s M
 		return s, err
 	}
 
-	db, err := database.New(ctx, privateDir)
+	db, err := database.New(ctx, privateDir, func(d string) { _ = cloudInit.RemoveDistroData(d) })
```

Comment:
It may make your tests just a little bit harder, but I think we should log the error here. Imagine later we have a problem with an instance being surprisingly initialised from cloud-init files seating in the folder for no known reason. This log line would tell when the clean-up failed, leading to that situation. We can pass to the log call the same context used to construct the database.

Reply:
I had misread `RemoveDistroData` and thought it was logging already, but it's only a decorate :), so good catch.